packages/profiling-node/src/integration.ts

```diff
@@ -24,8 +31,296 @@ function addToProfileQueue(profile: RawThreadCpuProfile): void {
+      PROFILE_QUEUE.splice(profileIndex, 1);
+      const profile = createProfilingEvent(client, cpuProfile, profiledTransaction);
+
+      if (client.emit && profile) {
+        const integrations =
+          client['_integrations'] && client['_integrations'] !== null && !Array.isArray(client['_integrations'])
+            ? Object.keys(client['_integrations'])
+            : undefined;
+
+        // @ts-expect-error bad overload due to unknown event
+        client.emit('preprocessEvent', profile, {
+          event_id: profiledTransaction.event_id,
+          integrations,
+        });
+      }
+
+      if (profile) {
+        profilesToAddToEnvelope.push(profile);
+      }
+    }
+
+    addProfilesToEnvelope(envelope, profilesToAddToEnvelope);
+  });
+}
+
+class ContinuousProfiler {
+  private _profilerId = uuid4();
+  private _client: NodeClient | undefined = undefined;
+
+  private _chunkId: string | undefined = undefined;
+  private _chunkTimer: NodeJS.Timeout | undefined = undefined;
+  private _chunkIntervalMS = 5000;
+  private _chunkStartTimestampMS: number | undefined = undefined;
+  private _chunkStartTraceID: string | undefined = undefined;
+
+  /**
+   * Called when the profiler is attached to the client (continuous mode is enabled). If of the profiler
+   * methods called before the profiler is initialized will result in a noop action with debug logs.
+   * @param client
+   */
+  public initialize(client: NodeClient): void {
+    this._client = client;
+  }
+
+  /**
+   * Recursively schedules chunk profiling to start and stop at a set interval.
+   * Once the user calls stop(), the current chunk will be stopped and flushed to Sentry and no new chunks will
+   * will be started. To restart continuous mode after calling stop(), the user must call start() again.
+   * @returns void
+   */
+  public start(): void {
+    if (!this._client) {
+      // The client is not attached to the profiler if the user has not enabled continuous profiling.
+      // In this case, calling start() and stop() is a noop action.The reason this exists is because
+      // it makes the types easier to work with and avoids users having to do null checks.
+      DEBUG_BUILD && logger.log('[Profiling] Profiler was never attached to the client.');
+      return;
+    }
+    if (this._chunkId || this._chunkTimer) {
+      DEBUG_BUILD &&
+        logger.log(
+          `[Profiling] Chunk with chunk_id ${this._chunkId} is still running, current chunk will be stopped a new chunk will be started.`,
+        );
+      this.stop();
+    }
+
+    const scope = getCurrentScope();
+    const isolationScope = getIsolationScope();
+
+    // Extract the trace_id from the current scope and assign start_timestamp as well as the new chunk_id.
+    this._chunkId = uuid4();
+    this._chunkStartTraceID = {
+      ...isolationScope.getPropagationContext(),
+      ...scope.getPropagationContext(),
+    }.traceId;
+    this._chunkStartTimestampMS = timestampInSeconds();
```

Comment:
h: Considering these get all assigned at the same time, could we make them part of a single object?

(doesn't make sense to have a set `_chunkId` without `_chunkStartTimestampMS`, but currently the types allow for this.